docs/language/control-flow.md

```diff
@@ -112,6 +113,78 @@ if let number = noNumber {
 }
 ```
 
+## Switch
+
+Switch-statements compare a value against several possible values of the same type, in order.
+When an equal value is found, the associated block of code is executed.
+
+The switch-statement starts with the `switch` keyword, followed by the tested value,
+followed by the cases inside opening and closing braces.
+The test expression must be equatable.
+The braces are required and not optional.
+
+Each case is a separate branch of code execution
+and starts with the `case` keyword,
+followed by a possible value, a colon (`:`),
+and the block of code that should be executed
+if the case's value is equal to the tested value.
+
+The block of code associated with a switch case
+[does not implicitly fall through](#no-implicit-fallthrough),
+and must contain at least one statement.
+Empty blocks are invalid.
+
+An optional default case may be given by using the `default` keyword.
+The block of code of the default case is executed
+when none of the previous case tests succeeded.
+It must always appear last.
+
+```cadence
+fun word(_ n: Int): String {
+    // Test the value of the parameter `n`
+    switch n {
+    case 1:
+        // If the value of variable `n` is equal to `1`,
+        // then return the string "one"
+        return "one"
+    case 2:
+        // If the value of variable `n` is equal to `1`,
+        // then return the string "two"
+        return "two"
+    default:
+        // If the value of variable `n` is neither equal to `1` nor to `2`,
+        // then return the string "other"
+        return "other"
+    }
+}
+```
+
+### `break`
+
+The block of code associated with a switch case may contain a `break` statement.
+It ends the execution of the switch statement immediately
+and transfers control to the code after the switch statement
+
+### No Implicit Fallthrough
+
+Unlike switch statements in some other languages,
+switch statements in Cadence do not "fall through":
+execution of the switch statement finishes as soon as the block of code
+associated with the first matching case is completed.
```

Comment:
We could use an example to explain the no-implicit fallthrough

```
// word(2) => "The number is two"
fun word(_ n: Int): String {
    let str = "The number is:"
    switch n {
    case 1:
        str = str + "one"
    case 2:
        str = str + "two"    
    default:
        str = str + "other"
    }
    return str
}
```

Do we have a similar tests for this case?

Reply:
Added a test and documentation in 8e3688bb5438c4fa32096263db2525f8b3cd92c5